File: java/src/main/java/com/example/api/ExampleApi.java
package com.example.api;

import net.corda.core.messaging.CordaRPCOps;

import javax.ws.rs.GET;
import javax.ws.rs.PUT;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.concurrent.ExecutionException;

// This API is accessible from /api/example. The endpoint paths specified below are relative to it.
@Path("example")
public class ExampleApi {
    private final CordaRPCOps services;

    public ExampleApi(CordaRPCOps services) {
        this.services = services;
    }

    /**
     * Define GET endpoints as follows.
     */
    @GET
    @Path("exampleGetEndpoint")
    @Produces(MediaType.APPLICATION_JSON)
    public void exampleGetEndpoint() {}

    /**
     * Define PUT endpoints as follows.
     */
    @PUT
    @Path("examplePutEndpoint")
    public Response createPurchaseOrder(Object payload) throws InterruptedException, ExecutionException {
        return Response.accepted().build();
    }
}

File: kotlin-source/src/main/kotlin/com/example/api/ExampleApi.kt
package com.example.api

import net.corda.core.messaging.CordaRPCOps
import javax.ws.rs.GET
import javax.ws.rs.PUT
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

// This API is accessible from /api/example. The endpoint paths specified below are relative to it.
@Path("example")
class ExampleApi(val services: CordaRPCOps) {
    /**
     * Define GET endpoints as follows.
     */
    @GET
    @Path("exampleGetEndpoint")
    @Produces(MediaType.APPLICATION_JSON)
    fun exampleGetEndpoint() {}

    /**
     * Define PUT endpoints as follows.
     */
    @PUT
    @Path("examplePutEndpoint")
    fun examplePutEndpoint(payload: Any): Response {
        return Response.accepted().build()
    }
}